pkg/ctl/create/cluster.go

```diff
@@ -94,9 +97,58 @@ func createClusterCmd() *cobra.Command {
 
 	fs.BoolVarP(&ng.PrivateNetworking, "node-private-networking", "P", false, "whether to make initial nodegroup networking private")
 
+	groupFlagsInUsage(cmd)
+
 	return cmd
 }
 
+func groupFlagsInUsage(cmd *cobra.Command) {
+	// Group flags by their categories determined by name prefixes
+	groupToPatterns := map[string][]string{
+		"Node":       {"node", "storage-class", "ssh", "max-pods-per-node", "full-ecr-access", "asg-access"},
+		"Networking": {"vpc", "zones",},
+		"Stack":      {"region", "tags"},
+		"Other":      {},
+	}
+	groups := []string{}
+	for k := range groupToPatterns {
+		groups = append(groups, k)
+	}
+	groupToFlagSet := make(map[string]*pflag.FlagSet)
+	for _, g := range groups {
+		groupToFlagSet[g] = pflag.NewFlagSet(g, /* Unused. Can be anythng. */ pflag.ContinueOnError)
+	}
+	cmd.LocalFlags().VisitAll(func(f *pflag.Flag) {
+		for _, g := range groups {
+			for _, p := range groupToPatterns[g] {
+				if strings.HasPrefix(f.Name, p) {
+					groupToFlagSet[g].AddFlag(f)
+					return
+				}
+			}
+		}
+		groupToFlagSet["Other"].AddFlag(f)
+	})
+
+	// The usage template is based on the one bundled into cobra
+	// https://github.com/spf13/cobra/blob/1e58aa3361fd650121dceeedc399e7189c05674a/command.go#L397
+	origFlagUsages := `
```

Comment:
Let's use https://github.com/lithammer/dedent here

Reply:
Umm, the whole purpose of this seemingly redundant empty lines is to make necessary and enough space between the usage` and the flags section.

That is, if I just wrap it with Decent like `origFlagUsages := Decent(` .... `)` what you get becomes:

```
Usage:
  eksctl create cluster [flags]



General flags:
  -n, --name string           EKS cluster name (generated if unspecified, e.g. "adorable-monster-1543548069")
  -r, --region string         AWS region
      --zones strings         (auto-select if unspecified)
      --tags stringToString   A list of KV pairs used to tag the AWS resources (e.g. "Owner=John Doe,Team=Some Team") (default [])

Initial nodegroup flags:
      --max-pods-per-node int     maximum number of pods per node (set automatically if unspecified)
```

Probably this isn't your intention?

Reply:
Sorry, I didn't notice that. The reason I was suggesting this is to avoid braking indentation, as currently the string has no indentation, and it's in a middle of a function, so it jumps straight at me when I look at it. Perhaps this is just my OCD :)